CMake pthread lookup breaking build

One commit breaks the build for me (Gentoo, gcc 4.9.4).
```
indi@astro1:~/indi_dev/indi$ git bisect bad
aee3c73c1278819c8a893cc27091b6eb229991c9 is the first bad commit
commit aee3c73c1278819c8a893cc27091b6eb229991c9
Author: Csaba Kertesz <csaba.kertesz@gmail.com>
Date:   Sat Jun 10 13:34:58 2017 +0300

    Add verbose warning and optimized debug/linker flags

:040000 040000 b9fccc43f30ab6b2eceb5931fb781c72e19b20f1 312da5aa41ff1597b76dbe189e0aae73581e4a92 M      libindi
```

PThread suddenly not found as a dependency.

```
+ cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=RelWithDebInfo /home/indi/indi_dev/indi/libindi/
-- The C compiler identification is GNU 4.9.4
-- The CXX compiler identification is GNU 4.9.4
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - not found
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
  Could NOT find Threads (missing: Threads_FOUND)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:388 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindThreads.cmake:223 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:201 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/indi/indi_dev/build/indi/CMakeFiles/CMakeOutput.log".
See also "/home/indi/indi_dev/build/indi/CMakeFiles/CMakeError.log".
```

I'll check how I can work around.
-Eric

--- a/libindi/CMakeLists.txt
+++ b/libindi/CMakeLists.txt
@@ -9,6 +9,7 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
 include(GNUInstallDirs)
 include(FeatureSummary)
+include(CheckCCompilerFlag)
 
 if(ANDROID OR "${CMAKE_SYSTEM_NAME}" STREQUAL "Android")
   set(ANDROID ON)
@@ -34,7 +35,16 @@ ENDIF ()
 
 # Add security (hardening flags)
 IF (UNIX OR APPLE OR ANDROID)
-    SET(SEC_COMP_FLAGS "-D_FORTIFY_SOURCE=2 -fstack-protector-all -fPIE")
+    # Older compilers are predefining _FORTIFY_SOURCE, so defining it causes a
+    # warning, which is then considered an error. Second issue is that for
+    # these compilers, _FORTIFY_SOURCE must be used while optimizing, else
+    # causes a warning, which also results in an error. And finally, CMake is
+    # not using optimization when testing for libraries, hence breaking the build.
+    CHECK_C_COMPILER_FLAG("-Werror -D_FORTIFY_SOURCE=2" COMPATIBLE_FORTIFY_SOURCE)
+    IF (${COMPATIBLE_FORTIFY_SOURCE})
+        SET(SEC_COMP_FLAGS "-D_FORTIFY_SOURCE=2")
+    ENDIF ()
+    SET(SEC_COMP_FLAGS "${SEC_COMP_FLAGS} -fstack-protector-all -fPIE")
     # Make sure to add optimization flag. Some systems require this for _FORTIFY_SOURCE.
     IF (NOT CMAKE_BUILD_TYPE MATCHES "MinSizeRel" AND NOT CMAKE_BUILD_TYPE MATCHES "Release" AND NOT CMAKE_BUILD_TYPE MATCHES "Debug")
         SET(SEC_COMP_FLAGS "${SEC_COMP_FLAGS} -O1")

